ui/src/message_popup/author/edit_question_page.jsx

```diff
@@ -14,12 +16,49 @@ export default React.createClass({
     questionId: React.PropTypes.string.isRequired
   },
 
+  getInitialState(){
+    return ({
+      loaded: false,
+      allDatabaseQuestions: {
+        currentQuestions: allQuestions,
+        archivedQuestions: allArchivedQuestions
+      }
+    });
+  },
+
+  componentDidMount(){
+    Api.questionsQuery().end(this.onQuestionsReceived);
+  },
+
+  onQuestionsReceived(err, response){
+    const questions = JSON.parse(response.text).questions;
+    if(questions !== undefined){
+      this.setState({loaded: true, allDatabaseQuestions: questions});
+      return;
+    }
+    this.setState({loaded: true});
+  },
+
   render(){
-    const question = withIndicator(withStudents([_.find(allQuestions, question => question.id.toString() === this.props.questionId)])[0]);
+    const {loaded, allDatabaseQuestions} = this.state;
+    const allQuestionsFlat =  this.state.allDatabaseQuestions.currentQuestions.concat(this.state.allDatabaseQuestions.archivedQuestions);
+    const originalQuestion = _.find(withStudents(allQuestionsFlat).map(question => withIndicator(question)), question => question.id.toString() === this.props.questionId);
     return (
-      <QuestionPage 
-        originalQuestion={question}
-        />
-      );
+      <div>
+        {!loaded &&
+          <QuestionPage
```

Comment:
The reasoning behind the `loaded` variable was to make the transition between the pages more fluid and not show any white screens (even though they may quickly disappear) while transitioning and waiting for the questions from the database.

It gets passed down so that `QuestionPage` can render its app bar and then wait for the questions from the database. Once the questions are received, it will load the rest of the components which get animated in so the transition looks smooth.